README.md

```diff
@@ -1,24 +1,150 @@
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+#### Original design idea by **[Gregoire Vella on Behance](https://www.behance.net/gregoirevella)**, in association with the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) of the design.
+<br />
 
-Things you may want to cover:
+# GoSave
 
-* Ruby version
+A mobile web application where you can manage your budget: you have a list of transactions associated with a category, so that you can see how much money you spent and on what.
 
-* System dependencies
+## Built With
 
-* Configuration
+- Ruby on Rails
 
-* Database creation
+## Heroku
+https://save-budget-app.herokuapp.com/
 
-* Database initialization
+## Video
+https://www.loom.com/share/2d68f7ca3b0d4a7ba6f26c5cbaa3e9b5
 
-* How to run the test suite
+## Additional Tools
 
-* Services (job queues, cache servers, search engines, etc.)
+- rubocop
+- stylelint
+- Ruby Gems
 
-* Deployment instructions
+## `Getting Started`
 
-* ...
+To get a local copy of this project:
+
+Clone this repository or download the Zip folder:
+```
+$ git clone git@github.com:KDlamini/Budget-app.git
+```
+Then:
+```
+$ cd Budget-app
+
+$ gem install bundler
+
+$ bundle install
+```
+
+To start App:
+```
+$ rails server
+```
+
+To view on browser:
+```
+http://localhost:3000
+```
+
+## `Database creation`
+
+Create a Postgres user:
+```
+$ su - postgres
+```
+
+After that access Postgres:
+```
+psql
+```
+
+Then create a user (or a “role”, as Postgres calls it):
+```
+CREATE ROLE budget WITH CREATEDB LOGIN PASSWORD 'budget';
+```
+
+Initialize the database:
+```
+$ bin/rails db:setup
+```
+
+Migrate the database:
+```
+$ bin/rails db:create
+```
+
+## `Tests`
+
+### Running specs:
+Default: Run all spec files (i.e., those matching spec/**/*_spec.rb):
+```
+$ bundle exec rspec
+```
+Run all spec files in a single directory (recursively):
+```
+$ bundle exec rspec spec/models
+```
+Run a single spec file:
+```
+$ bundle exec rspec spec/models/user_spec.rb
+```
+
+Run a single example from a spec file (by line number):
+```
+$ bundle exec rspec spec/models/user_spec.rb:8
+```
+
+See all options for running specs:
+```
+$ bundle exec rspec --help
+```
+
+### rails tests:
+
+```
+$ bin/rails test
+```
+
+### To track linter errors locally follow these steps:
+
+Track Ruby linter errors run:
+```
+$ rubocop
+```
+To auto-correct correctable Rubocop offenses run:
+```
+$ rubocop --auto-correct-all | rubocop -A
+```
+
+## `Authors`
+
+👤 **Ayoub Chahir**
+
+- GitHub: [@AyoubMS](https://github.com/AyoubMs)
+- Twitter: [@KnowledgeUserD](https://twitter.com/KnowledgeUserD)
+- LinkedIn: [Ayoub Chahir](https://www.linkedin.com/in/ayoub-chahir/)
+
+## 🤝 Contributing
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](../../issues/).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
```

Comment:
- [ ] OPTIONAL
It's a nice and smart thing to give appropriate credit to **[Gregoire Vella on Behance](https://www.behance.net/gregoirevella)**, the author of [the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/). Kindly ensure to add this under the **Acknowledgements** section of your ReadMe.